### PR TITLE
Minor fix

### PR DESCRIPTION
**Pull request by biyuhao**
Fix string comparison
Fix possible index out of bound

### CODE REVIEW

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ GovernanceConfiguration getDynamicConfiguration() {
     Registry getRegistry() {
         Registry registry = null;
         if (registryUrl == null) {
-            if (StringUtils.isNotEmpty(registryAddress)) {
+            if (StringUtils.isBlank(registryAddress)) {
```
Comment (lovepoem) [inline dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java]: Change the function ?   I think it should be `StringUtils.isNotBlank `

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java [hunk 2]
```diff
@@ -129,11 +129,7 @@ MetaDataCollector getMetadataCollector() {
     }
 
     private URL formUrl(String config, String group) {
-        String protocol = config.split("://")[0];
-        String address = config.split("://")[1];
-        String port = address.split(":")[1];
-        String host = address.split(":")[0];
-        URL url = new URL(protocol, host, Integer.parseInt(port));
+        URL url = URL.valueOf(config);
         if (StringUtils.isNotEmpty(group)) {
```
Comment (lovepoem) [inline dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java]: I think it should be `StringUtils.isNotBlank `

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/service/impl/RouteServiceImpl.java [hunk 3]
```diff
@@ -339,14 +339,14 @@ private String parseCondition(List<String> conditions) {
                 String[] array = condition.split("=>", 2);
                 String consumer = array[0].trim();
                 String provider = array[1].trim();
-                if (consumer != "") {
+                if (consumer.length() != 0) {
```
Comment (lovepoem) [inline dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/service/impl/RouteServiceImpl.java]: I think it should be `StringUtils.isNotBlank `

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/service/impl/RouteServiceImpl.java [hunk 4]
```diff
@@ -339,14 +339,14 @@ private String parseCondition(List<String> conditions) {
...
                     if (when.length() != 0) {
                         when.append(" & ").append(consumer);
                     } else {
                         when.append(consumer);
                     }
                 }
-                if (provider != "") {
+                if (provider.length() != 0) {
```
Comment (lovepoem) [inline dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/service/impl/RouteServiceImpl.java]: I think it should be `StringUtils.isNotBlank `

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java (hunk 1 above)
Comment (biyuhao) [reply dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java]: From the exception info blow, I think the logic here is meant to check both `registrlUrl` and `registryAddress ` are empty.

File: dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java (hunk 2 above)
Comment (biyuhao) [reply dubbo-admin-backend/src/main/java/org/apache/dubbo/admin/config/ConfigCenter.java]: We are building a `org.apache.dubbo.common.URL` object,
if we use `URL#valueOf()` when input string is null or malformed we got `throw new IllegalArgumentException("url == null");`.
Shall we leave the Inspection to `valueOf` internal here?

**Comment by nzomkxia** [comment]
Thank you for your pr, I'll merge it

**Pull request merged**